src/libraries/System.Runtime.Serialization.Xml/ref/System.Runtime.Serialization.Xml.cs

```diff
@@ -6,6 +6,50 @@
 
 namespace System.Runtime.Serialization
 {
+    public abstract partial class DataContract
+    {
+        public DataContract? BaseContract { [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")] get { throw null; } }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")]
+        public DataContract BindGenericParameters(DataContract[] paramContracts, System.Collections.Generic.Dictionary<DataContract, DataContract> boundContracts) { throw null; }
+        public string? ContractType { get { throw null; } }
+        internal DataContract(DataContractCriticalHelper helper) { }
+        internal const System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes DataContractPreserveMemberTypes =
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicMethods |
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.NonPublicMethods |
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicConstructors |
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.NonPublicConstructors |
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicFields |
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicProperties;
+        public static string EncodeLocalName(string localName) { throw null; }
+        public GenericInfo? GenericInfo { [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")] get { throw null; } }
```

Comment:
It looks like this method can be duplicated in the schema code and doesn't need to be exposed.

Reply:
EncodeLocalName absolutely can be done out of box. But it feels like the responsibility of DataContract, and I prefer one source of truth if possible, so I left it there and exposed it. If it gets pushback in the API review, we can always come back to this and duplicate it in the schema package.